text/0033-logging.md

```diff
@@ -0,0 +1,189 @@
+
+A rolling log strategy offers permits to:
+
+- Facilitate the administration of systems that generate a large number of logs.
+- It automates swapping, compressing, deleting, and sending logs. It assists in keeping the logging system within the specified file system space limits.
+- Defines an interval over which log analysis can be performed
+- Gives an efficient way to identify log files that are no longer used so that an automated process can clean up and compress the log directory and run log analysis programs.
+
+[See more information here about log rotation strategy](https://en.wikipedia.org/wiki/Log_rotation)
+
+This is something important to have when systems generate a lot of logs or are under heavy load. It is also a great practice to feed a monitoring dashboard at certain times and, through this, to offer an easy way to analyze and understand what has happened in the system in a human-readable way.
+
+Internally, Elasticsearch uses log4j2 to trace events and configure the log strategy. Logging levels can be configured per package, giving the precision needed to monitor a specific feature or time within the lifecycle of a search engine.
+
+Log level can be: `debug`, `info`, `warn`, `error`, `fatal`, or `unknown`.
+
+#### Elastic Entreprise Search
+
+Elasticsearch's SaaS platform, like Algolia, offers a quite similar endpoint to browse the logs generated by the system.
+
+However, it is possible to deactivate the storage of logs, modify the retention period but also to filter the logs with more details.
+
+![image](https://www.elastic.co/guide/en/app-search/current/images/guides/log-settings-controls.png)
+*Elastic App Search's Log Retention settings view.*
+
+> It is also possible to do that from a Log Settings API endpoint.
+
+> API Logs is aimed for trace requests and responses at engine level while Analytics Logs trace query, clicks and counts.
+
+### IV. Explanation
+
+#### Current Logging behaviour of MeiliSearch (0.20)
+
+MeiliSearch uses `env_logger` to allow setting the output level of logs from the `RUST_LOG` environment variable. `env_logger` permits to set a specific log level per module if needed. [See more here](https://docs.rs/env_logger/0.8.3/env_logger/).
+
+#### Logging behaviour for Milli (0.21)
+
+We have decided to keep the use of `env_logger` for Milli/Transplant. However, we are going to make changes to make the logging more consistent and more versatile.
+
+##### Log Levels
+
+| Level | Description                                                                                                              |
+|-------|--------------------------------------------------------------------------------------------------------------------------|
+| ERROR | Everything related to an error occuring.                                                                                         |
+| WARN  | Everything related to a warning. TBD                                                                                     |
```

Comment:
I need to know what is the `TBD` before opening an issue 🙂

Reply:
Or maybe we don't add warnings for the v0.21.0? I don't know if the spec concerns only the v0.21.0 or the future releases as well